packages/realtime_client/lib/src/realtime_client.dart

```diff
@@ -387,13 +407,17 @@ class RealtimeClient {
 
   /// communication has been closed
   void _onConnClose() {
-    final event = conn?.closeReason ?? '';
+    final statusCode = conn?.closeCode;
+    RealtimeCloseEvent? event;
+    if (statusCode != null) {
+      event = RealtimeCloseEvent(code: statusCode, reason: conn?.closeReason);
+    }
     log('transport', 'close', event);
 
     /// SocketStates.disconnected: by user with socket.disconnect()
     /// SocketStates.closed: NOT by user, should try to reconnect
     if (connState == SocketStates.closed) {
-      _triggerChanError(event);
+      _triggerChanError();
```

Comment:
Removing the `event` here to align the behavior with the JS client
https://github.com/supabase/realtime-js/blob/master/src/RealtimeClient.ts#L388

Reply:
I'm unsure about this change. This breaks the current heartbeat timeout test. This causes a channel error to be emitted, but without any reason.

Reply:
>This causes a channel error to be emitted, but without any reason.

This is the behavior of the JS SDK, so we are aligning the behavior there.